ci-operator/step-registry/baremetalds/sno/rename/baremetalds-sno-rename-commands.sh

```diff
@@ -0,0 +1,317 @@
+
+rename_script=$(cat <<IEOF
+#!/usr/bin/env bash
+
+set -euoE pipefail
+
+export KUBECONFIG=/etc/kubernetes/static-pod-resources/kube-apiserver-certs/secrets/node-kubeconfigs/localhost.kubeconfig
+function wait_for_api {
+  echo "Waiting for API..."
+  until oc get clusterversion &> /dev/null
+  do
+    echo "Waiting for API..."
+    sleep 5
+  done
+  echo "API is available"
+}
+
+function start_containers {
+  systemctl start crio
+  systemctl start kubelet
+}
+
+function stop_containers {
+  systemctl stop kubelet
+  crictl ps -q | xargs crictl stop || true
+  crictl rmp -a -f || true
+  systemctl stop crio
+}
+
+function recert {
+  local release_image="${RELEASE_IMAGE:-quay.io/openshift-release-dev/ocp-release:4.14.2-x86_64}"
+  local etcd_image="\$(oc adm release extract --from="\${release_image}" --file=image-references | jq '.spec.tags[] | select(.name == "etcd").from.name' -r)"
+  local recert_image="${RECERT_IMAGE:-quay.io/edge-infrastructure/recert:latest}"
+  local previous_base_domain="${PREVIOUS_BASE_DOMAIN:-redhat.com}"
+  local previous_cluster_name="${PREVIOUS_CLUSTER_NAME:-test-infra-cluster}"
+  local base_domain="${BASE_DOMAIN:-another.domain}"
+  local cluster_name="${CLUSTER_NAME:-another-name}"
+
+  podman run --authfile=/var/lib/kubelet/config.json \
+      --name recert_etcd \
+      --detach \
+      --rm \
+      --network=host \
+      --privileged \
+      --entrypoint etcd \
+      -v /var/lib/etcd:/store \
+      "\${etcd_image}" \
+      --name editor \
+      --data-dir /store \
+
+  sleep 10
+
+  podman run -it --network=host --privileged \
+      -v /tmp/certs:/certs  \
+      -v /tmp/keys:/keys \
+      -v /etc/kubernetes:/kubernetes \
+      -v /var/lib/kubelet:/kubelet \
+      -v /etc/machine-config-daemon:/machine-config-daemon \
+      -v /etc/cni/multus:/multus \
+      -v /var/lib/ovn-ic:/ovn-ic \
+      \${recert_image} \
+      --etcd-endpoint localhost:2379 \
+      --static-dir /kubernetes \
+      --static-dir /kubelet \
+      --static-dir /machine-config-daemon \
+      --static-dir /multus \
+      --static-dir /ovn-ic \
+      --use-cert /certs/admin-kubeconfig-client-ca.crt \
+      --use-key "kube-apiserver-localhost-signer /keys/localhost-serving-signer.key" \
+      --use-key "kube-apiserver-lb-signer /keys/loadbalancer-serving-signer.key" \
+      --use-key "kube-apiserver-service-network-signer /keys/service-network-serving-signer.key" \
+      --use-key "\${ROUTER_CA_CN} /keys/router-ca.key" \
+      --cn-san-replace api-int.\${previous_cluster_name}.\${previous_base_domain}:api-int.\${cluster_name}.\${base_domain} \
+      --cn-san-replace api.\${previous_cluster_name}.\${previous_base_domain}:api.\${cluster_name}.\${base_domain} \
+      --cn-san-replace *.apps.\${previous_cluster_name}.\${previous_base_domain}:*.apps.\${cluster_name}.\${base_domain} \
+      --cluster-rename \${cluster_name}:\${base_domain} \
+
+  podman kill recert_etcd
+}
+
+function fetch_crts_keys {
```

Comment:
nit: Can we place this function higher up? And in general order the function definitions in the order they're called, makes it easier to follow